go/glome/glome.go

```diff
@@ -231,6 +231,19 @@ func (d *Dialog) Tag(msg []byte, counter uint8) []byte {
 // to the correct tag. Also, its length must be at least MinPeerTagLength
 // and always smaller than MaxTagSize.
 func (d *Dialog) Check(tag []byte, msg []byte, counter uint8) bool {
+	expect := d.expect(tag, msg, counter)
+	return hmac.Equal(expect, tag)
+}
+
+// CheckEncoded method checks if an encoded tag matches some provided message and the counter.
+// The used encoding is base64-url. For more details look at the Check method description.
+func (d *Dialog) CheckEncoded(tag string, msg []byte, counter uint8) bool {
+	expect := d.expect([]byte(tag), msg, counter)
+	return base64.URLEncoding.EncodeToString(expect) == tag
```

Comment:
hmac.Equal?

Reply:
`hmac.Equal()` works only with byte arrays, while here I compare strings. I guess that is the only way to check encoded tags when the received one is truncated and it's bit length (without padding) is not divided by 8 (like in the [first test vector](https://github.com/google/glome/blob/master/docs/glome-login.md#vector-1), where the response tag is truncated to the ten-letter word and is represented with 60bits).

Reply:
Resolving the conversation as there it is not actual anymore.